src/control.rs

```diff
@@ -370,19 +508,36 @@ mod tests {
         // finish the archive and unwrap it as a byte vector
         let archive_bytes = in_ar.into_inner().unwrap();
 
-        // parse the archive bytes
+        // check that the expected files were included in the archive
         let mut out_ar = tar::Archive::new(&archive_bytes[..]);
 
-        // compare the file names in the archive to what we expect
         let mut archived_file_names = decode_names(&mut out_ar);
         archived_file_names.sort();
 
-        // don't check the file content, generation of correct files and content
-        // is tested at a lower level, we're only testing the higher level.
-        let expected_maintainer_scripts = vec!["postinst", "postrm", "prerm"]
+        let mut expected_maintainer_scripts = maintainer_scripts
             .iter()
-            .map(|i| i.to_string())
+            .map(|(script, _)| filename_from_path_str(script))
             .collect::<Vec<String>>();
+        expected_maintainer_scripts.sort();
+
         assert_eq!(expected_maintainer_scripts, archived_file_names);
+
+        // check the content of the archived files for any unreplaced placeholders.
+        // create a new tar wrapper around the bytes as you cannot seek the same
+        // Archive more than once.
+        let mut out_ar = tar::Archive::new(&archive_bytes[..]);
+
+        let unreplaced_placeholders = out_ar
+            .entries()
+            .unwrap()
+            .map(Result::unwrap)
+            .map(|mut entry| {
+                let mut v = String::new();
+                entry.read_to_string(&mut v).unwrap();
```

Comment:
It doesn't matter for tests, but FYI there's `std::fs::read_to_string`.

Reply:
I thought I was forgetting something, but it was late and a quick search didn't find that, thanks!

Reply:
Hmm, `std::fs::read_to_string()` only works for something `Path` like, but in this case we have an `Entry` from the `tar` crate. There's also [`std::io::read_to_string()`](https://doc.rust-lang.org/std/io/fn.read_to_string.html#) but that's only in nightly Rust.

Reply:
Oh, you're right. I haven't noticed it's an entry.